director/director_test.go

```diff
@@ -1729,19 +1729,20 @@ func TestRedirectMiddleware(t *testing.T) {
 
 }
 func TestRedirects(t *testing.T) {
+	server_utils.ResetTestState()
 	ctx, cancel, egrp := test_utils.TestContext(context.Background(), t)
-	defer func() { require.NoError(t, egrp.Wait()) }()
```

Comment:
It actually doesn't look like the `cancel` or `egrp` get used anywhere in this test, other than to be cleaned up at the end. Can you change this to:
```
ctx, _, _ := test_utils.TestContext(context.Background(), t)
```
to clean up some of the test logic?